saleor/core/tasks.py

```diff
@@ -16,18 +18,15 @@ def delete_from_storage_task(path):
 @app.task
 def delete_event_payloads_task():
     delete_period = timezone.now() - settings.EVENT_PAYLOAD_DELETE_PERIOD
-    deliveries = EventDelivery.objects.filter(created_at__lte=delete_period)
-    attempts = EventDeliveryAttempt.objects.filter(
-        Q(Exists(deliveries.filter(id=OuterRef("delivery_id"))))
-        | Q(delivery__isnull=True, created_at__lte=delete_period)
-    )
-    payloads = EventPayload.objects.filter(
-        deliveries__isnull=True, created_at__lte=delete_period
-    )
-
-    attempts._raw_delete(attempts.db)
-    deliveries._raw_delete(deliveries.db)
-    payloads._raw_delete(payloads.db)
+    valid_deliveries = EventDelivery.objects.filter(created_at__gt=delete_period)
+    payloads_to_delete = EventPayload.objects.filter(
+        ~Exists(valid_deliveries.filter(payload_id=OuterRef("id")))
+    ).order_by('-pk')
+    ids = payloads_to_delete.values_list("pk", flat=True)[:BATCH_SIZE]
+    qs = EventPayload.objects.filter(pk__in=ids)
+    if ids:
+        qs.delete()
+        delete_event_payloads_task.delay()
```

Comment:
Give up if it takes too long, e.g. if the first task started 1h ago, then give up (1h is a random value out of my head).

We don't want to end-up in a case where we have concurrent delete tasks against the same rows, we will end-up with a lock queue. On top of not wanting to run forever a single task.